### ISSUE TITLE
Add document explaining the authentication system

### ISSUE THREAD

**Issue by davidfowl** [aspnet/Docs#4055]
## Requests for new Topics

> Write a **short** paragraph describing what the topic will cover.

Today our authentication documentation is scenario focused. This is fine but doesn't help explain the fundamentals of the system and the various component responsibilities. We should have an introduction to the authentication system that describes the services, the verbs (challenge, authentication, forbid, signin, signout), what authentication schemes are etc. There should also be diagrams of a typical authentication flow with all of the actors involved.

> 2. Write an outline of the proposed topic contents.

Here's an example of what I think this topic could look like:
https://digitalmccullough.com/posts/aspnetcore-auth-system-demystified.html

> 3. Where would you put the topic in the Table of Contents.

https://docs.microsoft.com/en-us/aspnet/core/security/authentication/

> 4. If you know who should review the issue, notify them by including @githubID in the issue description.

@HaoK @blowdart @Tratcher

**Comment by HaoK** [comment]
@Rick-Anderson is there someone on the doc side that I can work with for this new topic?

**Comment by Rick-Anderson** [comment]
@scottaddie  can you work with Haok on this?

**Comment by davidfowl** [comment]
@HaoK I'd like these sorts of samples as part of the doc https://github.com/leastprivilege/AspNetCoreSecuritySamples/tree/aspnetcore2.

**Comment by HaoK** [comment]
@davidfowl how do you feel about Authentication.Cookies/JWT/etc in entropy as the official home for the samples that the documents will reference?

**Comment by Tratcher** [comment]
No, the docs have their own copy of the samples they reference.

**Comment by HaoK** [comment]
Ok, but I'll still put MVC + auth sample reference apps similar to Domincs in entropy, since the security ones can't depend on MVC

**Comment by Tratcher** [comment]
We've also had asks for a demo on dynamically adding and removing auth schemes.

**Comment by Tratcher** [comment]
See https://github.com/aspnet/Security/blob/0959c941b40086b131b1e01d304fc23bb887fdc8/test/Microsoft.AspNetCore.Authentication.Test/DynamicSchemeTests.cs#L137-L143

**Comment by HaoK** [comment]
@scottaddie we went ahead and created an official home for the AuthSamples, they will be slowly coming online here: https://github.com/aspnet/AuthSamples

We can tailor the samples towards what this doc needs as well.  We still need to figure out how to deal with the keys/cred issue for the auth providers that require external setup (OAuth/OIDC).

**Comment by HaoK** [comment]
@scottaddie a new sample showing how to dyamically add schemes and update their options has been added here: https://github.com/aspnet/AuthSamples/tree/dev/samples/DynamicSchemes

**Comment by Rick-Anderson** [comment]
@scottaddie  how about doing this is phases? 

* Create a page that links to the samples
* Create article.

**Comment by scottaddie** [comment]
@Rick-Anderson I'm okay with that approach.

**Comment by Rick-Anderson** [comment]
See Add some documentation for claims transformation (Auth) #2097

**Comment by Rick-Anderson** [comment]
Per @JamesNK  in #13764

There is no article that gives an overview of authentication in ASP.NET Core.

The article should talk about:

What authentication is
What authentication options there are
An introduction to setting up authentication
It should be located at: Security and Identity -> Authentication -> Overview

Equivalent page for authorization: https://docs.microsoft.com/en-us/aspnet/core/security/authorization/introduction
